actions/hdl_example/sw/Makefile

```diff
@@ -27,8 +27,19 @@
 
 # This is solution specific. Check if we can replace this by generics too.
 
+# Finding $SNAP_ROOT
+ifndef SNAP_ROOT
+# check if we are in sw folder of an action (three directories below snap root)
+ifneq ("$(wildcard ../../../ActionTypes.md)","")
+SNAP_ROOT=$(abspath ../../../)
+else
+$(info You are not building your software from the default directory (/path/to/snap/actions/<action_name>/sw) or specified a wrong $$SNAP_ROOT.)
+$(error Please source /path/to/snap/hardware/snap_settings.sh or set $$SNAP_ROOT manually.)
```

Comment:
The name of the script for checking and setting up the environment variables is `snap_settings` (without `.sh`).
Furthermore, the new config management (see #416) will introduce other methods to set up `$SNAP_ROOT`.
Maybe the message should just say: "Please make sure that $$SNAP_ROOT is set up correctly".
This comment applies to the corresponding error messages contained in other modified `sw/Makefile` files as well.